component/core/src/main/java/org/exoplatform/social/core/metadata/storage/MetadataStorage.java

```diff
@@ -137,15 +137,29 @@ public int countMetadataItemsByMetadataTypeAndCreator(long metadataType, long cr
   }
 
   public Map<String, Long> countMetadataItemsByMetadataTypeAndAudienceId(long metadataType, long creatorId, long spaceId) {
-    List<Tuple> metadataItemsTuple =
-                                   metadataItemDAO.countMetadataItemsByMetadataTypeAndAudienceId(metadataType, creatorId, spaceId);
+    List<Tuple> metadataItemsTuple = metadataItemDAO.countMetadataItemsByMetadataTypeAndAudienceId(metadataType,
+                                                                                                   creatorId,
+                                                                                                   spaceId);
     Map<String, Long> metadataItemsMap = new HashMap<>();
     for (Tuple tuple : metadataItemsTuple) {
       metadataItemsMap.put((String) tuple.get(0), ((BigInteger) tuple.get(1)).longValue());
     }
     return metadataItemsMap;
   }
 
+  public Map<Long, Long> countMetadataItemsByMetadataTypeAndSpacesIdAndCreatorId(long metadataType,
+                                                                                 long creatorId,
+                                                                                 List<Long> spacesId) {
+    List<Tuple> metadataItemsTuple = metadataItemDAO.countMetadataItemsByMetadataTypeAndSpacesIdAndCreatorId(metadataType,
```

Comment:
The Tuple object is specific to JPA and thus must be processed in DAO layer only.

```suggestion
    List<Long, Long> metadataCountBySpace = metadataItemDAO.countMetadataItemsByMetadataTypeAndSpacesIdAndCreatorId(metadataType,
```

Reply:
fixed!